atlasdb-autobatch/src/main/java/com/palantir/atlasdb/autobatch/DisruptorAutobatcher.java

```diff
@@ -61,11 +61,7 @@ private static ThreadFactory threadFactory(String safeLoggablePurpose) {
     }
 
     private static ThreadFactory createThreadFactory(String safeLoggablePurpose) {
-        String namePrefix = String.format("autobatcher.%s-", safeLoggablePurpose);
-        return new ThreadFactoryBuilder()
-                .setDaemon(true)
-                .setNameFormat(namePrefix + "%d")
-                .build();
+        return new NamedThreadFactory("autobatcher." + safeLoggablePurpose, true);
```

Comment:
Flag that for internal ski product, this will basically report weird metrics? We start many TransactionManagers, so if you hoped this would give you a total count and whatnot, don't think this will work.

Reply:
That said I don't think it's necessarily worse than no-instrumentation, just don't know if this will get us what we want. Also, this will be a few extra time series.

Reply:
Actually, had a look at the metrics, and looks like this will be cumulative, so all good.

Reply:
The underlying metric objects are shared so duplicate names are okay, if the safeLoggablePurpose differs that gives me even more specificity. Either way it solves my problem :-)